src/coreclr/src/jit/compiler.cpp

```diff
@@ -4226,7 +4230,8 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
         return;
     }
 
-    assert(!compDonotInline());
+    // By this point we are only jitting the root method (and not an inlinee)
```

Comment:
Minor: I might reword this as "By this point we have completed inlining into the root method, so going forward we won't encounter any inlinees" or something like that.

Reply:
At this point we haven't started inlining into the root method.

We run a few of the initial phases on inlinees. So inlinees and root methods both call `compCompile`. This comment is trying to say that "at this point in the phase list, all the inlinee phases have been run, and inlinee compiles have exited, so we should only get this far if we are jitting the root method."

If that quoted text seems better, let me know, and I'll update...

Reply:
Yes, that would be great - and thanks for clarifying!